app/src/services/api/api.ts

```diff
@@ -24,6 +46,30 @@ export const api = generatedApi.enhanceEndpoints({
     destroySource: {
       invalidates: (_, { id }) => [{ type: "Sources", id }],
     },
+    listResources: {
+      provides: (response) => [
+        ...response.map(({ id }) => ({ type: "Resources" as "Resources", id })),
```

Comment:
"Resources" as "Resources" ?

Reply:
Typescript has some issues infering the type here so I need to cast the type attribute... [According to the Redux team](https://discord.com/channels/102860784329052160/103538784460615680/811333139565379605) this issue may or may not appear depending on the Typescript version. "Resources" is an [entity type](https://rtk-query-docs.netlify.app/api/createapi/#entitytypes) declared above (in addEntityType) and used for [revalidation](https://rtk-query-docs.netlify.app/concepts/mutations#advanced-mutations-with-revalidation).